File: library-manager-ui/src/services/cartService.js
import { axiosPrivate } from '~/apis/configHttp';

export const addToCart = (bookId) => {
    return axiosPrivate.post(`carts/add?bookId=${bookId}`);
};

export const getCartDetails = () => {
    return axiosPrivate.get('carts/details');
};

export const removeFromCart = (id) => {
    return axiosPrivate.delete(`carts/remove?cartDetailId=${id}`);
};

export const clearCart = () => {
    return axiosPrivate.delete('carts/clear');
};

File: library-manager-server/src/main/java/com/example/librarymanager/service/CartService.java
package com.example.librarymanager.service;

import com.example.librarymanager.domain.dto.pagination.PaginationFullRequestDto;
import com.example.librarymanager.domain.dto.pagination.PaginationResponseDto;
import com.example.librarymanager.domain.dto.response.BorrowRequestSummaryResponseDto;
import com.example.librarymanager.domain.dto.response.CommonResponseDto;
import com.example.librarymanager.domain.dto.response.GetCartDetailResponseDto;

import java.util.List;

public interface CartService {

    List<GetCartDetailResponseDto> getCartDetails(String cardNumber);

    CommonResponseDto addToCart(String cardNumber, Long bookId);

    CommonResponseDto removeFromCart(String cardNumber, Long cartDetailId);

    CommonResponseDto clearCart(String cardNumber);

    PaginationResponseDto<BorrowRequestSummaryResponseDto> getPendingBorrowRequests(PaginationFullRequestDto requestDto);
}
